src/Admin/AbstractAdmin.php

```diff
@@ -1582,7 +1583,10 @@ public function getRoot()
             return $this;
         }
 
-        return $this->getParentFieldDescription()->getAdmin()->getRoot();
+        $admin = $this->getParentFieldDescription()->getAdmin()->getRoot();
+        \assert($admin instanceof self);
```

Comment:
can you show the error if you don't add this?

Reply:
It would be nice to avoid using assert indeed

Reply:
yes, you are correct.

The error was
```
  1588   Method Sonata\AdminBundle\Admin\AbstractAdmin::getRoot() should return Sonata\AdminBundle\Admin\AbstractAdmin but returns Sonata\AdminBundle\Admin\AdminInterface.  

```

it was since `AbstractAdmin` and `AdminInterface` had different return types for `getRoot` method


`AbstractAdmin::getRoot` -> `AbstractAdmin`
`AdminInterface::getRoot` -> `AdminInterface`

I've fixed it